src/components/validations/form-with-validations.hoc.js

```diff
@@ -63,10 +68,11 @@ const withValidations = (WrappedComponent) => {
         <ValidationsContext.Provider value={ this.getContext() }>
           <WrappedComponent
             validate={ this.validateRegisteredInputs }
-            validationsCount={ this.state.validationsCount }
-            warningsCount={ this.state.warningsCount }
-            infoCount={ this.state.infoCount }
             isValidating={ this.state.formIsValidating }
+            { ...Object.keys(VALIDATION_TYPES).reduce((acc, type) => ({
+              ...acc,
+              [`${type}Count`]: this.state[`${type}Count`]
+            }), {}) }
```

Comment:
I think it could be more readable if you create a separate reducer function. It takes a while to grasp what happens here.